### ISSUE TITLE
Process params by copy and not by reference

### ISSUE THREAD

**Issue by claustres** [jb55/s3-blob-store#19]
The `params` passed in the options are modified in place https://github.com/jb55/s3-blob-store/blob/master/index.js#L34. This leads to an issue when you call the `createWriteStream` method twice with the same object. Here is the pseudo-code to reproduce the issue:
```
let params = { ACL: 'public-read' }
store.createWriteStream({ key: '1', params })
// Here the new key will be ignored and '1' will be used instead because params has been mutated
// and now contains a Key entry = '1' that overrides the provided key as an option
store.createWriteStream({ key: '2', params })
```

I don't know if it can affect `createReadStream` as well.

**Comment by bitflower** [comment]
I have the same issue. Any chance we can fix this? Will you accept PRs?

**Comment by jb55** [comment]
> I have the same issue. Any chance we can fix this? Will you accept PRs?

yea of course

**Comment by bitflower** [comment]
Hi @jb55 !

Hope you're fine. Mind releasing a new version including the fix?

Thanks!

**Comment by jb55** [comment]
@bitflower this was included in v4.0 which was released 2 years ago

**Comment by bitflower** [comment]
Sorry for that. I had looked for the lodash dep which was replaced in the meantime.
